dask/array/tests/test_percentiles.py

```diff
@@ -9,19 +9,35 @@
 
 def test_percentile():
```

Comment:
I'd write this with `pytest.mark.parametrize` over the method arg:

```python
@pytest.mark.parametrize('method', ['tdigest', 'dask'])
def test_percentile(method):
    ...
```

For arguments that don't support one type I'd just special case around them inside the test:

```python
    if method != 'tdigest':
        x = np.array(['a', 'a', 'd', 'd', 'd', 'e'])
        d = da.from_array(x, chunks=(3,))
        assert_eq(da.percentile(d, [0, 50, 100]),
                  np.array(['a', 'd', 'e'], dtype=x.dtype))
```

Doing it this way ensures we have coverage across both methods and minimizes duplicating code.

Reply:
That's a really nice way of doing it, thanks for pointing that out!